web/Controllers/PropertyController.cs

```diff
@@ -11,52 +10,107 @@ namespace BricksAndHearts.Controllers;
     }
 
-    [Authorize(Roles = "Landlord")]
+    [Authorize(Roles = "Landlord, Admin")]
     [HttpGet("add/step/{step:int}")]
-    public ActionResult AddNewProperty_Continue([FromRoute] int step)
+    public ActionResult AddNewProperty_Continue([FromRoute] int step, int propertyId, int? landlordId = null)
+    {
+        var newPropertyModel = new PropertyViewModel { Address = new PropertyAddress() };
+        landlordId ??= GetCurrentUser().LandlordId!.Value;
+
+        if (propertyId != 0)
+        {
+            var property = _propertyService.GetPropertyByPropertyId(propertyId);
+            if (property == null)
+            {
+                return RedirectToAction("ViewProperties", "Landlord", new { id = landlordId });
+            }
+
+            newPropertyModel = PropertyViewModel.FromDbModel(property);
+        }
+        else
+        {
+            newPropertyModel.PropertyId = propertyId;
+            newPropertyModel.LandlordId = (int)landlordId;
+        }
+
+        if (!(GetCurrentUser().IsAdmin || GetCurrentUser().LandlordId == newPropertyModel.LandlordId))
+        {
+            return StatusCode(403);
+        }
+
+        // Show the form for this step
+        return View("AddNewProperty", new AddNewPropertyViewModel { Step = step, Property = newPropertyModel });
+    }
+
+    [Authorize(Roles = "Landlord, Admin")]
+    [HttpGet("edit/step/{step:int}")]
+    public ActionResult EditProperty_Continue([FromRoute] int step, int propertyId)
     {
-        var landlordId = GetCurrentUser().LandlordId!.Value;
-        
-        var dbModel = _propertyService.GetIncompleteProperty(landlordId);
-        var property = dbModel == null
-            ? new PropertyViewModel { Address = new PropertyAddress() }
-            : PropertyViewModel.FromDbModel(dbModel);
-        
-        return View("AddNewProperty", new AddNewPropertyViewModel { Step = step, Property = property });
+        var property = _propertyService.GetPropertyByPropertyId(propertyId);
+        if (property == null)
+        {
+            return StatusCode(404);
+        }
+
+        var newPropertyModel = PropertyViewModel.FromDbModel(property);
+        newPropertyModel.PropertyId = propertyId;
+
+        // Show the form for this step
+        if (!(GetCurrentUser().IsAdmin || GetCurrentUser().LandlordId == newPropertyModel.LandlordId))
+        {
+            return StatusCode(403);
+        }
+
+        return View("EditProperty", new AddNewPropertyViewModel { Step = step, Property = newPropertyModel });
     }
 
-    [Authorize(Roles = "Landlord")]
+    [Authorize(Roles = "Landlord, Admin")]
     [HttpPost("add/step/{step:int}")]
-    public async Task<ActionResult> AddNewProperty_Continue([FromRoute] int step, [FromForm] PropertyViewModel newPropertyModel)
+    public async Task<ActionResult> AddNewProperty_Continue([FromRoute] int step, int propertyId,
+        [FromForm] PropertyViewModel newPropertyModel, int? landlordId = null)

```

Comment:
My gut feeling is no, but now I'm thinking about it, it probably shouldn't. I'm going to have a go at removing it quickly and see if anything breaks :P

Reply:
It appears to be yes, despite removing it as a route value, and assigning it to the model, and then passing it back through a hidden field, the landlord ID returns to 0 each time. 
So it has to be reassigned here from the route value to have it persist through the initial steps of the process, before it is retrievable from the database. 

I'd also have preferred the routing to be different, using the /propertyId/landlordid/ style as opposed to queries, but it might be another ticket to solve this?

Reply:
I think I've found how to fix this issue with it returning to zero each time, using a different input from"@html.hiddenfor"

Reply:
Sounds good 👍 

I agree it would be nicer to have it in the path rather than queries, but not a big deal, let's get this merged!